types/block.go

```diff
@@ -1324,11 +1324,6 @@ func (data *Data) Hash() cmtbytes.HexBytes {
 	return data.hash
 }
 
-// GetDataRootHash returns the hash data which is not equal to calling data.Hash()
-func (data *Data) GetDataRootHash() cmtbytes.HexBytes {
-	return data.hash
-}
-
```

Comment:
[question] why was this removed? This PR replaces usage of `.GetDataRootHash()` with `data.Hash()` which seems sus given:

> which is not equal to calling data.Hash()

Reply:
the nil check makes it equivalent. i can push to the other pr, accidentally pushed to this one